File: passenger/src/main/java/com/modsen/passenger/service/PassengerService.java
package com.modsen.passenger.service;

import com.modsen.passenger.dto.request.PassengerCreate;
import com.modsen.passenger.dto.request.PassengerUpdate;
import com.modsen.passenger.dto.response.PassengerResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PassengerService {

    PassengerResponse findById(Integer id);

    Page<PassengerResponse> findAll(Pageable pageable);

    PassengerResponse update(Integer id, PassengerUpdate request);

    void deleteById(Integer id);

    PassengerResponse create(PassengerCreate request);
}

File: passenger/src/main/kotlin/com/modsen/ktpassenger/service/PassengerService.kt
package com.modsen.ktpassenger.service

import com.modsen.ktpassenger.dto.PassengerCreate
import com.modsen.ktpassenger.dto.PassengerResponse
import com.modsen.ktpassenger.dto.PassengerUpdate
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface PassengerService {

    fun findById(id: Int): PassengerResponse

    fun findAll(pageable: Pageable): Page<PassengerResponse>

    fun update(id: Int, request: PassengerUpdate): PassengerResponse

    fun deleteById(id: Int)

    fun create(request: PassengerCreate): PassengerResponse
}